project1/src/main/java/uk/co/cpsd/javaproject1/World.java

```diff
@@ -1,22 +1,37 @@
 package uk.co.cpsd.javaproject1;
 
+import java.io.FileWriter;
+import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Stream;
 
+
 public class World {
     public final int size = 10;
     private List<Animal> animals;
     private int totalTicks;
-    private final int MAX_GRASS_AGE=25;
+    private final int MAX_GRASS_AGE=100;
     private int[][] grassDeathTime= new int[size][size]; 
+    private List<Integer> goatHistory = new ArrayList<>();
+    private List<Integer> grassHistory = new ArrayList<>();
     public World(int numOfGoats){
         animals=new ArrayList<>();
         for(int i=0;i<numOfGoats;i++){
             animals.add(new Goat((int)(Math.random() * size),(int)(Math.random() * size)));
         }
     }
 
+    public List<Integer> getGoatHistory() {
+        return goatHistory;
```

Comment:
C# is so much better than Java at this.  You can have public List<integer> goatHistory { get; private set; }
But you can't do that in Java, so carry on :)